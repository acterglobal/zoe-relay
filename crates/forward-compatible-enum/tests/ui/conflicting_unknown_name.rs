use forward_compatible_enum::ForwardCompatibleEnum;

#[derive(ForwardCompatibleEnum)]
#[forward_compatible(unknown_variant = "Variant1")]  // Conflicts with existing variant!
pub enum ConflictingUnknownName {
    #[discriminant(1)]
    Variant1,
    Unknown { discriminant: u32, data: Vec<u8> },
}