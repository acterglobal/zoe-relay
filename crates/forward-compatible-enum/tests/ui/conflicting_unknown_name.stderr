error: Unknown variant 'Variant1' must be a struct variant with 'discriminant: u32' and 'data: Vec<u8>' fields
 --> tests/ui/conflicting_unknown_name.rs:6:5
  |
6 | /     #[discriminant(1)]
7 | |     Variant1,
  | |____________^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/conflicting_unknown_name.rs:9:2
  |
9 | }
  |  ^ consider adding a `main` function to `$DIR/tests/ui/conflicting_unknown_name.rs`
